src/components/OSCALMetadata.js

```diff
@@ -38,17 +53,181 @@ const OSCALMetadataPartiesCard = styled(Paper)(({ theme }) => ({
   backgroundColor: theme.palette.background.paper,
   position: "relative",
   overflow: "auto",
-  maxHeight: "12em",
+  maxHeight: "18em",
 }));
 
 // Returns a string with a locality-sensitive representation of this date
 const formatDate = (isoDate) => new Date(isoDate).toLocaleString();
 
+const TELEPHONE_ICON_MAPPING = {
+  home: <HomeIcon />,
+  mobile: <SmartphoneIcon />,
+  office: <BusinessIcon />,
+};
+
+const ADDRESS_ICON_MAPPING = {
+  home: <HomeIcon />,
+  work: <BusinessIcon />,
+};
+
+const UNKNOWN_TYPE_ICON = <HelpOutlineIcon />;
+
+function getIconOfType(mapping, contactType) {
+  if (!mapping || !contactType) {
+    return UNKNOWN_TYPE_ICON;
+  }
+  return mapping[contactType] ?? UNKNOWN_TYPE_ICON;
+}
+
+function getPhoneIcon(contactType) {
+  return getIconOfType(TELEPHONE_ICON_MAPPING, contactType);
+}
+
+function getAddressIcon(contactType) {
+  return getIconOfType(ADDRESS_ICON_MAPPING, contactType);
+}
+
+export function OSCALMetadataPartyAddress(props) {
+  const { address } = props;
+  return (
+    <>
+      {getAddressIcon(address.type)}
+      <address>
+        <Typography>
+          {`${address["addr-lines"]?.join(", ")} ${address.city} ${
+            address["postal-code"]
+          } ${address.country}`}
+        </Typography>
+      </address>
+    </>
+  );
+}
```

Comment:
```suggestion

export function OSCALMetadataPartyAddress(props) {
  const { address } = props;
  const addrString = [
    ...address["addr-lines"],
    `${address.city}, ${address.state} ${address["[postal-code"]}`,
    address.country
  ].filter(line => line).join("\n");
  return (
    <>
      {getAddressIcon(address.type)}
      <address>
        <Typography>
        {addrString}
        </Typography>
      </address>
    </>
  );
}
```

With better formatting, variable names, and error handling.